packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -185,23 +186,25 @@ const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
     }
 
     // TODO: fix type discrepancy between WithAuthOptions and AuthenticateRequestOptions
-    const requestState = await authenticateRequest(req, options as AuthenticateRequestOptions);
-    const requestStateHeaders = requestState.headers;
+    const authenticateRequestOptions = {
+      ...options,
+      secretKey: options.secretKey || SECRET_KEY,
+      publishableKey: options.publishableKey || PUBLISHABLE_KEY,
+      ...handleMultiDomainAndProxy(req, options as AuthenticateRequestOptions),
+    } as AuthenticateRequestOptions;
+    const requestState = await clerkClient.authenticateRequest(req, authenticateRequestOptions);
+    const locationHeader = requestState.headers?.get('location');
 
-    const locationHeader = requestStateHeaders?.get('location');
-
-    // triggering a handshake redirect
     if (locationHeader) {
-      return new Response(null, { status: 307, headers: requestStateHeaders });
+      // triggering a handshake redirect
```

Comment:
Okay so I realize that I misspoke earlier...after looking at the logic, we only redirect when the status is handshake, so the check you had initially added is the right approach 👍